tenzoMath/tenzoMath.cpp

```diff
@@ -1,149 +1,521 @@
+        fgmaxCurr.at<double>(2, 2) = _fgmax.at<double>(2, 2) * gravProjection.at<double>(2, 0);
+
+        tmaxCurr.at<double>(2, 0) = _tmax.at<double>(2, 0) * gravProjection.at<double>(2, 0);
+        tmaxCurr.at<double>(2, 1) = _tmax.at<double>(2, 1) * gravProjection.at<double>(2, 0);
+        tmaxCurr.at<double>(2, 2) = _tmax.at<double>(2, 2) * gravProjection.at<double>(2, 0);
+    }
+    else
+    {
+        fgmaxCurr.at<double>(2, 0) = -_fgmaxNeg.at<double>(2, 0) * gravProjection.at<double>(2, 0);
+        fgmaxCurr.at<double>(2, 1) = -_fgmaxNeg.at<double>(2, 1) * gravProjection.at<double>(2, 0);
+        fgmaxCurr.at<double>(2, 2) = -_fgmaxNeg.at<double>(2, 2) * gravProjection.at<double>(2, 0);
+
+        tmaxCurr.at<double>(2, 0) = -_tmaxNeg.at<double>(2, 0) * gravProjection.at<double>(2, 0);
+        tmaxCurr.at<double>(2, 1) = -_tmaxNeg.at<double>(2, 1) * gravProjection.at<double>(2, 0);
+        tmaxCurr.at<double>(2, 2) = -_tmaxNeg.at<double>(2, 2) * gravProjection.at<double>(2, 0);
+    }
+
+    newData[0] = rawData[0] - _forcesBias[0] - fgmaxCurr.at<double>(0, 0) - fgmaxCurr.at<double>(1, 0) - fgmaxCurr.at<double>(2, 0);
+    newData[1] = rawData[1] - _forcesBias[1] - fgmaxCurr.at<double>(0, 1) - fgmaxCurr.at<double>(1, 1) - fgmaxCurr.at<double>(2, 1);
+    newData[2] = rawData[2] - _forcesBias[2] - fgmaxCurr.at<double>(0, 2) - fgmaxCurr.at<double>(1, 2) - fgmaxCurr.at<double>(2, 2);
+    newData[3] = rawData[3] - _torquesBias[0] - tmaxCurr.at<double>(0, 0) - tmaxCurr.at<double>(1, 0) - tmaxCurr.at<double>(2, 0);
+    newData[4] = rawData[4] - _torquesBias[1] - tmaxCurr.at<double>(0, 1) - tmaxCurr.at<double>(1, 1) - tmaxCurr.at<double>(2, 1);
+    newData[5] = rawData[5] - _torquesBias[2] - tmaxCurr.at<double>(0, 2) - tmaxCurr.at<double>(1, 2) - tmaxCurr.at<double>(2, 2);
+
+    return newData;
+}
+
+void TenzoMath::loadCalibData()
+{
+    std::ifstream input("calibData.txt");
+    if (!input)
+        std::cout << "file is not found\n";
+    input >> _forcesBias[0] >> _forcesBias[1] >> _forcesBias[2] >> _torquesBias[0] >> _torquesBias[1] >> _torquesBias[2];
+    for (int i = 0; i < 3; ++i)
+        for (int j = 0; j < 3; ++j)
+            input >> _fgmax.at<double>(i, j);
+    for (int i = 0; i < 3; ++i)
+        for (int j = 0; j < 3; ++j)
+            input >> _fgmaxNeg.at<double>(i, j);
+    for (int i = 0; i < 3; ++i)
+        for (int j = 0; j < 3; ++j)
+            input >> _tmax.at<double>(i, j);
+    for (int i = 0; i < 3; ++i)
+        for (int j = 0; j < 3; ++j)
+            input >> _tmaxNeg.at<double>(i, j);
+    input.close();
+}
+
+void TenzoMath::calculatePos(std::array<double, 6>& curPos, std::array<double, 6> ftReadings)
+{
+    cv::Mat currRot = FanucModel::rotMatrix(curPos[3] / 180.0 * PI, curPos[4] / 180.0 * PI,
+       curPos[5] / 180.0 * PI);
+
+    constexpr double coefForces = 0.005;
+    constexpr double coefTorques = 0.001;
+    constexpr double threshold = 150;
```

Comment:
(**note**): Wow! You know about `constexpr` variables!) So use it in every place you can.